admin/tabs/activity/tracking/charts/ChartsPanel.js

```diff
@@ -7,32 +7,69 @@
 import {isEmpty} from 'lodash';
 import {uses, hoistCmp} from '@xh/hoist/core';
 import {panel} from '@xh/hoist/desktop/cmp/panel';
-import {tabContainer} from '@xh/hoist/cmp/tab';
-import {toolbar} from '@xh/hoist/desktop/cmp/toolbar';
-import {switchInput} from '@xh/hoist/desktop/cmp/input';
+import {toolbar, toolbarSeparator} from '@xh/hoist/desktop/cmp/toolbar';
+import {button} from '@xh/hoist/desktop/cmp/button';
+import {buttonGroupInput} from '@xh/hoist/desktop/cmp/input';
+import {Icon} from '@xh/hoist/icon';
+import {chart} from '@xh/hoist/cmp/chart';
+import {checkbox} from '@xh/hoist/desktop/cmp/input';
+
 import {ChartsModel} from './ChartsModel';
 
 export const chartsPanel = hoistCmp.factory({
     model: uses(ChartsModel),
     render({model}) {
-        if (isEmpty(model.dimensions)) return null;
+        const {chartType, dimensions} = model,
+            isTimeSeries = chartType == 'timeseries';
+
+        if (isEmpty(dimensions)) return null;
         return panel({
-            item: tabContainer(),
-            bbar: bbar(),
-            model: {
-                defaultSize: 500,
-                side: 'bottom',
-                prefName: 'xhAdminActivityChartSize'
-            }
+            items: [
+                chart({
+                    model: model.categoryChartModel,
+                    omit: isTimeSeries
```

Comment:
This would be much more straightforward with a ternary on the model and a single .  But I wonder if you would need to also assign the key of the component to the id of the model

Reply:
I tried to do just that that but ran into issues. 

It seemed that the model switching wasn't being accepted by the chart for some reason. (It would just stay with the type of chart I had originally designated, even if the render function ran IIRC) Perhaps the chart itself wasn't rerendering despite the switch? Anyway I immediately knew this would work so I didn't really did any further.

Reply:
yes, need to just assign the key to the component to get it to re-render when the model pulled out from under it.  but this is fine as well